java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/searcher/JUnit5TestSearcher.java

```diff
@@ -151,6 +114,29 @@ public boolean findAnnotation(IAnnotationBinding[] annotations, String[] annotat
         return false;
     }
 
+    @Override
+    public boolean isTestClass(IType type) throws JavaModelException {
+        return JUNIT5_TEST_FINDER.isTest(type);
+    }
+
+    @Override
+    public TestItem[] findTestsInContainer(IJavaElement element, IProgressMonitor monitor) throws CoreException {
+        final Map<String, TestItem> result = new HashMap<>();
+        final Set<IType> types = new HashSet<>();
+        JUNIT5_TEST_FINDER.findTestsInContainer(element, types, monitor);
+        for (final IType type : types) {
+            final TestItem item = TestItemUtils.constructTestItem(type, TestLevel.CLASS, TestKind.JUnit5);
+            item.setChildren(Arrays.stream(type.getMethods())
+                    .map(m -> String.format("%s@%s", m.getJavaProject().getProject().getName(),
```

Comment:
`String.format("%s@%s"` is much slower. Because the pattern is simple, suggest to use  `string + "@" + string`, which will be optimized by JVM to use StringBuilder under the hood.